service/controller/resource/spark/create.go

```diff
@@ -100,6 +100,14 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 				ObjectMeta: metav1.ObjectMeta{
 					Name:      secretName(sparkCR.Name),
 					Namespace: azureMachinePool.Namespace,
+					OwnerReferences: []metav1.OwnerReference{
+						{
+							APIVersion: sparkCR.APIVersion,
+							Kind:       sparkCR.Kind,
+							Name:       sparkCR.Name,
+							UID:        sparkCR.UID,
+						},
```

Comment:
You might want to use foreground deletion by setting `BlockOwnerDeletion` to `true` here. This way the `Spark` CR won't be deleted until the corresponding `Secret` is deleted if I understand correctly.

Reply:
Good idea. Done.